### PR TITLE
Change Lite.pm and ImageMagick-6-policy.xml via patches in Dockerfile (and another small change)

### PR DESCRIPTION
**Pull request by drgrice1**
This add patches to `/usr/share/perl5/XMLRPC/Lite.pm` and `/etc/ImageMagick-6/policy.xml`, and applies those at build time in Dockerfile, instead of importing the full files.

I also added a bit of tweaking of the Dockerfile.  Set the args `WEBWORK2_GIT_URL`, `WEBWORK2_BRANCH`, `PG_GIT_URL`, and `PG_BRANCH` in docker-compose.yml instead of in the Dockerfile.  Also, don't convert those to environment variables.  That isn't needed.  This requires that `docker-compose build` is used rather than `docker build .`.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -64,35 +64,34 @@ services:
...
+        #
+        #  context: /Path_To/webwork2/
+        #  dockerfile: /Path_To/Dockerfile
+        args:
+            # build args specifying the branches for webwork2 and pg used to build the image
+            # To use the master branches of webwork2 and pg 
+            #- ARG WEBWORK2_BRANCH=master
+            #- ARG PG_BRANCH=master
+            # To use the 2.15 branches of webwork2 and pg from the "official" GitHub repositories
+            - WEBWORK2_GIT_URL=https://github.com/taniwallach/webwork2.git
```
Comment (taniwallach) [inline docker-compose.yml]: This should probably have `WEBWORK2_GIT_URL=https://github.com/openwebwork/webwork2.git` and the next line should use a regular branch and not the temporary `docker-to-ubuntu-2004` feature branch.

**Review by taniwallach — CHANGES_REQUESTED** [review]
I looked over the changes, but have not yet tested them out...
I don't think we want to provide a version of `docker-compose.yml` which actively refers to a temporary feature branch or to my fork instead of the main repository. Samples of such a type can be included.

File: docker-compose.yml (hunk 1 above)
Comment (drgrice1) [reply docker-compose.yml]: Yeah, oops.  That was there because I was using your branch for testing.

Comment (drgrice1) [reply docker-compose.yml]: I fixed this.

**Review by taniwallach — APPROVED** [review]
After the small correction, it all looks OK.

`docker-compose build` worked OK.

I had a startup time issue, as https://github.com/openwebwork/webwork2/pull/1160 was not yet merged, so is not in the code being used by the image created,  but the `docker-compose.yml` has `WEBWORK_DB_DRIVER: MariaDB` set. Swaping to `WEBWORK_DB_DRIVER: mysql` sufficed to allow testing to the extent of being able to log in to a course.

**Comment by taniwallach** [comment]
I tested for tonight only by working with a branch with these changes merged in... I have not directly retested the revised PR after these changes.
```
git fetch tani pull/17/head:test_glenn_Docker_17 
git checkout test_glenn_Docker_17
```
and then making a directory with the new `Dockerfile` and `docker-compose.yml`, making changes to use that location, timezone settings, an `image` name, and a few local config files.

**Comment by drgrice1** [comment]
> I had a startup time issue, as [openwebwork#1160](https://github.com/openwebwork/webwork2/pull/1160) was not yet merged, so is not in the code being used by the image created, but the `docker-compose.yml` has `WEBWORK_DB_DRIVER: MariaDB` set. Swaping to `WEBWORK_DB_DRIVER: mysql` sufficed to allow testing to the extent of being able to log in to a course.

Yeah, that is what I had to do also.  This will work though after openwebwork#1160 is merged.

**Pull request merged**